### DISCUSSION TITLE
More general buttons

### DISCUSSION THREAD

**Discussion by Incrementis** [NearInfinityBrowser/NearInfinity#175] [Ideas & Feature Requests]
hello,

when editing save files, changes cannot be saved. NI must be completely closed to save.
Right now I need to manipulate a character file for [rudimentary testing (e.g. switching scripts)](https://github.com/Incrementis/IEEE-Player-Character-Voice-Installer-/issues/4).

**Here is what I do:**
1. Navigate to folder _Save_
2. Select one _BALDUR.gam_ file
3. Go to tab _view_
4. Double click(or press _View/Edit_ button) one player character in the _Player Characters_ area
5. Select _CRE resource_ row in tab _Edit_
6. Press _View/Edit_ button
7. _CRE resurce_ window opens up
8. Manipulate file
9. Close NI to save changes

All of these steps must be performed for each individual change and test required. This can be tedious depending on the scope of the testing.

It would be helpful if NI could offer all the buttons that are available in other windows (see image as an example).
This would prevent the closure of the entire NI.
![grafik](https://github.com/NearInfinityBrowser/NearInfinity/assets/17119474/72058695-3a0a-4077-aa1a-690e364d0753)

Thanks for your time!

**Comment by Argent77** [comment]
Export and Save buttons are always available on the root window or view of an opened game resource. This is valid for all structured game resources, including BALDUR.GAM files.

By modifying CRE data of a (Non-)Player Character you are actually just modifying the CRE version embedded in the selected savegame file. It's comparable to modifying an ability structure in an ITM or SPL file. To save the changes you can switch back to the BALDUR.GAM view or window, select the Edit tab and press the Save button. Alternatively you can force a Save prompt by selecting another resource in the resource tree.

**Reply by Incrementis** [reply]
This reduces the steps, but not significantly, as each new change to the save file requires a new iteration of the steps.
The steps that are not necessary in your described case are:
1. Reopen NI
2. Navigate to folder _Save_ 

This is just a small improvement, but I understand that my use case may be a rarity (I can't really say).

**Comment by Argent77** [comment]
I can understand that it's tedious to navigate to the actual CRE data of a saved character since it involves two layers of substructures. I'll see if there is a way to improve it somewhat.

**Comment by Argent77** [comment]
2b9bcb5d92aa1e19a425d6f555359a951057102b should help you navigate quicker to GAM's CRE structures. The feature can be tested in the [nightly release](https://github.com/Argent77/NearInfinity/releases/tag/nightly).

**Reply by Incrementis** [reply]
Thank you for the effort!
I have now found time to deal with the new changes
I downloaded the nightly release JAR file, tried it out and unfortunately didn't notice any difference.

I'm not particularly familiar with Java, so if I understand it correctly on a surface level, it adds some new mouse events that might be needed for the View/Edit buttons(!?). These are added to the Viewer(GamResource gam) which may correspond to the **BALDUR.gam** file.
The code also adds some new "view/edit" buttons to the CHR and CRE files, which do something different, I'm not entirely sure.

Any help here would be greatly appreciated.
Thank you very much.

**Reply by Argent77** [reply]
Sorry, I should have explained the changes in my previous comment. The feature allows you to jump directly to the CRE resource structure if you right-click on a (non-)player character in the list and select to open the CRE structure.

<details>
<summary><strong>Screenshot</strong></summary>
<img src="https://github.com/NearInfinityBrowser/NearInfinity/assets/4519923/17253fcc-a1ce-4f78-9a0b-5fc0bc5cb253" alt="Screenshot"/>
</details>
